smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/xml/MemberShapeDecodeXMLGenerator.kt

```diff
@@ -296,12 +299,22 @@ abstract class MemberShapeDecodeXMLGenerator(
         if (member.hasTrait(SwiftBoxTrait::class.java)) {
             memberTargetSymbol = memberTargetSymbol.recursiveSymbol()
         }
-        val decodeVerb = if (memberTargetSymbol.isBoxed() && !isUnion) "decodeIfPresent" else "decode"
+        val decodeVerb = if (memberTargetSymbol.isBoxed() && !isUnion || (member.hasTrait<DefaultTrait>())) "decodeIfPresent" else "decode"
         val decodedMemberName = "${memberNameUnquoted}Decoded"
+
+        val defaultValNilCoalescing = member.getTrait(DefaultTrait::class.java).getOrNull()?.let { trait ->
+            val defaultVal = trait.toNode()
+            when {
+                defaultVal.isStringNode() -> "?? \"$defaultVal\""
+                defaultVal.toString().equals("null") -> "?? nil"
```

Comment:
Is there a better way to detect a null default value?  I am always wary of detecting a type by string like this... how does a null default actually get modeled by Smithy?

Reply:
That's a great point, I missed out on that. There exists a `isNullNode()` method for Smithy's Node types, so I've changed the code to use that method.

Reply:
https://smithy.io/2.0/spec/type-refinement-traits.html#smithy-api-default-trait

Now I'm sort of wondering if `null` is a valid default value:
> The `@default` trait on root-level shapes cannot be set to `null`.

(Not exactly sure what they mean by "root level")

Reply:
lol the answer is right above...

> The `@default` trait on a structure member can be set to `null` to explicitly indicate that the member has no default value or to override the default value requirement of a targeted shape.

so never mind, we do need the `null` case here.